provider/provider_factory.go

```diff
@@ -0,0 +1,48 @@
+package provider
```

Comment:
There will be a cyclic dependency between sdkv2 provider and plugin framework provider due to common parts such as tests so we would need to move sdkv2 provider code to internal as well to solve this properly. 

Doing that in this PR would make the changes big and I think it would be more readable to split them. 

I have moved most of the logic that is used by only plugin framework to internal/pluginframework/provider but for the rest there would need to be some refactor. Will do that up in followup PRs.